chaosazure/vmss/actions.py

```diff
@@ -242,8 +247,21 @@ def choose_vmss_instance_at_random(vmss_choice, configuration, secrets):
         logger.debug(
             "Found virtual machine scale set instances: {}".format(
                 [x['name'] for x in vmss_instances]))
-    logger.warning("Temporarily hard coding to the first instance which sould be zone 0")            
-    choice_vmss_instance = vmss_instances[0]
+    logger.warning("Temporarily hard coding to the first instance which sould be zone 0")
+    choice_vmss_instance = random.choice(vmss_instance)
+    return choice_vmss_instance
+
+def choose_vmss_instance(vmss_choice, configuration, secrets, vmss_name):
+    vmss_instances = fetch_vmss_instances(vmss_choice, configuration, secrets)
+    if not vmss_instances:
+        logger.warning("No virtual machine scale set instances found")
```

Comment:
it's a very good idea to give two distinct messages in cases like this:

a debug log with lots of information for debugging
a message that helps the narration of the experiment via teh exception